backend/typescript/services/implementations/notificationService.ts

```diff
@@ -0,0 +1,192 @@
+    try {
+      const notification = await prisma.notificationReceived.findUnique({
+        where: {
+          id,
+        },
+      });
+      if (!notification) throw new Error(`notification id ${id} not found`);
+
+      return notification;
+    } catch (error: unknown) {
+      Logger.error(
+        `Failed to get Notification. Reason = ${getErrorMessage(error)}`,
+      );
+      throw error;
+    }
+  }
+
+  async sendNotification(
+    authorId: number,
+    title: string,
+    message: string,
+    recipientIds: number[],
+  ): Promise<NotificationDTO> {
+    try {
+      const newNotification = await prisma.notification.create({
+        data: {
+          title,
+          message,
+          author: {
+            connect: { id: authorId },
+          },
+          recipients: {
+            create: recipientIds.map((recipient) => ({
+              recipient: {
+                connect: {
+                  id: recipient,
+                },
+              },
+            })),
+          },
+        },
+        include: {
+          recipients: true,
+        },
+      });
+
+      return newNotification;
+    } catch (error) {
+      Logger.error(
+        `Failed to create Notification. Reason = ${getErrorMessage(error)}`,
+      );
+      throw error;
+    }
+  }
+
+  async deleteUserNotification(
+    notificationId: number,
+  ): Promise<NotificationDTO> {
+    try {
+      const deletedNotification = await prisma.notification.delete({
+        where: {
+          id: notificationId,
+        },
+        include: { recipients: true },
+      });
+
+      if (!deletedNotification)
+        throw new Error(`notification id ${notificationId} not found`);
+
+      return deletedNotification;
+    } catch (error) {
+      Logger.error(
+        `Failed to set isDelete flag. Reason = ${getErrorMessage(error)}`,
+      );
+      throw error;
+    }
+  }
+
+  async updateSeenNotification(
+    notificationRecievedId: number,
+  ): Promise<NotificationRecievedDTO> {
+    try {
+      await prisma.notificationReceived.update({
+        where: {
+          id: notificationRecievedId,
+        },
+        data: {
+          seen: true,
+        },
+      });
+
+      const updatedNotification = await prisma.notificationReceived.findUnique({
+        where: {
+          id: notificationRecievedId,
+        },
+      });
+
+      if (!updatedNotification)
+        throw new Error(`notification id ${notificationRecievedId} not found`);
+
+      return updatedNotification;
+    } catch (error) {
+      Logger.error(
+        `Failed to set seen flag. Reason = ${getErrorMessage(error)}`,
+      );
+      throw error;
+    }
+  }
+
+  async sendAnnouncement(
+    title: string,
+    message: string,
+    userId: number,
+  ): Promise<NotificationDTO> {
+    let newNotification: NotificationDTO;
+    try {
+      // const activeResidents = await residentService.getActiveResidents();
```

Comment:
FYI, we simply used all users from UserStub to be active users to test and run send Announcements, in the actual code we should be using line 155 (which should return users post refactor) and line 157 can be removed